lib/Rex/Interface/Exec/Local.pm

```diff
@@ -119,6 +119,11 @@ sub exec {
 
   }
 
+  # we need to bitshift $? so that $? contains the right (and for all
+  # connection methods the same) exit code after a run()/i_run() call.
+  # this is for the user, so that he can query $? in his task.
+  $? >>= 8;
```

Comment:
No it doesn't modify it. The Net::SSH2::Channel object has a method `exit_value` (or something similar) which provides the "unbitshifted" return value. The overwirde of $? is done by rex.